### ISSUE TITLE
utc_strftime does not handle day-of-year (j) properly

### ISSUE THREAD

**Issue by varenius** [skyfielders/python-skyfield#456]
I think I found a bug in utc_strftime. Let's define a time in skyfield as:
```
ts = load.timescale()
t= ts.utc(2020, 9, 29, 10, 20, 0)
```
This seems to work and give a sensible timestamp:
```
print(str(t))
<Time tt=2459121.9313562964>
```
I now want to use the function utc_strftime to print this time as a string in a few different ways. First works fine:
```
print(t.utc_strftime('%Y.%m.%d.%H:%M:%S'))
2020.09.29.10:20:00
```
I also want to print this in a different format, using "day of year" (which is 273 for this date). According to https://rhodesmill.org/skyfield/api-time.html#skyfield.timelib.Time.utc_strftime the function utc_strftime "calls Python’s time.strftime() to format the date and time." and "You can find the full list, along with options that control field widths and leading zeros, at: https://docs.python.org/3/library/time.html#time.strftime". At this python-doc-page we see "%j | Day of the year as a decimal number [001,366]." so we should get 273 here. But we don't! Instead:
```
print(t.utc_strftime('%Y.%j.%H:%M:%S'))
2020.001.10:20:00
```
001 is wrong! It should be 273. If I try a quick workaround, we indeed find 273:
```
print(t.utc_datetime().strftime('%Y.%j.%H:%M:%S'))
2020.273.10:20:00
```
So, it seems that pythons "strftime" does work as intended, but skyfield's "utc_strftime" does not. I would expect the two to give the same result. Or am I missing something here?

**Comment by brandon-rhodes** [comment]
It’s indeed likely that Skyfield is not computing a day-of-year for its `strftime`-related routines. I'll take a look at the math involved, and either support it, or print a clear error instructing the user to build a real Python `datetime` and ask it for the day of the year.

What application, may I ask, are you building that needs that particular quantity? This is my first encounter with it in real code, so I'm interested to know!

**Comment by varenius** [comment]
In this particular instance, I was using skyfield to translate TLEs to other coordinates at particular times, to be put in SNAP files. SNAP (not easy to find a good reference online; there are other things named "snap" these days) is an ancient (70s?) control-file format used by some old telescope control systems, such as the Very Long Baseline Interferometry (VLBI) Field-System. The "day of year" is used in various VLBI-related tools and utilities, both for observations scheduling and data analysis. In SNAP files, the time tags are given as e.g. "2020.275.19:19:58.76".

The overlap between modern code such as skyfield, and ancient VLBI tools (such as field-system) is rare. I just needed some quick solution to convert TLEs, and I thought of using Skyfield since it is neat and compact. I thought I might as well output also the timestamps using Skyfield's strftime as well, and stumbled upon this bug. 

So: this really doesn't have to work, but an error (as you suggest) would be nice (as it now claims it can support all the strftime features).

**Comment by brandon-rhodes** [comment]
This is going to be a difficult computation in the case where the user requests a cutover from the old Julian Calendar to the new Gregorian Calendar, as that will create a year with an unusually short number of days — so I can't assume that the month-and-day will let me learn how many days preceded it in the year; I'm going to have to compute afresh when January 1 was, and then subtract. I'll see if I can get that done reasonably in the code!

**Comment by brandon-rhodes** [comment]
@varenius — I think I've landed a fix! Could you try out the development version to see how it works for you?
```
pip install https://github.com/skyfielders/python-skyfield/archive/master.zip
```
Thanks — and, thanks for detailing the application so I could see how the value is being used, it's helpful!

**Comment by brandon-rhodes** [comment]
I have just released 1.30 with this fix included. Let me know if you run into any problems with the new feature. Thanks!

**Comment by varenius** [comment]
Apologies for the late response - busy days.

I have verified that version 1.30 works as intended, thanks!
